src/utils.ts

```diff
@@ -303,3 +313,49 @@ export async function createUnsignedFillRelayTransaction(
     { from: simulatedRelayerAddress }
   );
 }
+
+/**
+ * Search back in time for the first block at or older than the desired lookback.
+ *
+ * The approach here is as below:
+ * 1. We use the skip distances, which are meant to be a couple of hours or less for each chain.
+ * 2. Fetch the first block one skip distance from the latest block. Check its timestamp to see how far back it is
+ * relatively to the latest block.
+ * 3. Apply a multiplier to the skip distance based on the time difference and find a block with the desired lookback
+ * This assumes the block production speed is roughly constant.
+ * 4. Check the timestamp again, if we still haven't achieved the desired lookback, repeat step 3 and go back further.
```

Comment:
nit: I think this algo repeats step 2 and goes on from there, not starting at step 3 right? Because step 2 first figures out the `lastBlockSpeed` which is important to approximate the desired block

Reply:
Updated